### PR TITLE
Set Up CI and Credentials Cleanup

### PR DESCRIPTION
**Pull request by omidbachari**
### Changes
1. [chore] Adds circle ci config to app. See https://circleci.com/docs/2.0/language-elixir/

2. [chore] Move developer-specific configs to `dev.secret.exs`

### Issues
Addresses https://github.com/pioneerworks/homebase-elixir/issues/3

### Outstanding TO-DO to Finish
- [x] Set up CI account
- [x] Add badge snippet to README

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+jobs:
+  build:
+    parallelism: 1
+    docker:
+      - image: circleci/elixir:1.7.3
+        environment:
+          MIX_ENV: test
+      - image: circleci/postgres:10.1-alpine
+        environment:
+          POSTGRES_USER: postgres
```
Comment (omidbachari) [inline .circleci/config.yml]: This came from the standard config.yml. I know the image comes with a DB and a user--but I wonder if user, db and password need to be updated.

Comment (skyeagle) [reply .circleci/config.yml]: Should be fine, let's see how it works

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    steps:
+      - checkout
+
+      - run: mix local.hex --force
+      - run: mix local.rebar --force
+
+      - restore_cache:
+          keys:
+            - v1-mix-cache-{{ .Branch }}-{{ checksum "mix.lock" }}
+            - v1-mix-cache-{{ .Branch }}
```
Comment (skyeagle) [inline .circleci/config.yml]: I guess we just need the key above only, I see no usage cases without mix.lock. what do you think?

Comment (omidbachari) [reply .circleci/config.yml]: Thanks, @skyeagle. Let me figure that out.

**Comment by omidbachari** [comment]
Yes, can we please add this to Circle @skyeagle?

**Comment by skyeagle** [comment]
@omidbachari I've set up it on CI, we need to merge this to provide a config. So, merge it whenever you are ready.

**Pull request merged**